src/main/host/process.rs

```diff
@@ -124,8 +126,20 @@ pub struct Process {
     total_run_time: Cell<Duration>,
 
     desc_table: RefCell<DescriptorTable>,
-    memory_manager: RefCell<Option<MemoryManager>>,
     itimer_real: RefCell<Timer>,
+
+    // References to `Self::memory_manager` cached on behalf of C code using legacy
+    // C memory access APIs.
+    // TODO: Remove these when we've migrated Shadow off of the APIs that need
+    // them (probably by migrating all the calling code to Rust).
+    //
+    // Must be before memory_manager for drop order.
+    unsafe_borrow_mut: RefCell<Option<UnsafeBorrowMut>>,
+    unsafe_borrows: RefCell<Vec<UnsafeBorrow>>,
+
+    // Must come after `unsafe_borrows` and `unsafe_borrow_mut`.
+    // Boxed to avoid invalidating those if Self is moved.
+    memory_manager: Box<RefCell<Option<MemoryManager>>>,
```

Comment:
Maybe mark these comments with "SAFETY" so that they're more noticable (in vim it's a highlighted word, I'm not sure about other editors)?

Reply:
Done